test/functional/test_framework/test_framework.py

```diff
@@ -273,6 +273,11 @@ def stop_nodes(self):
             # Wait for nodes to stop
             node.wait_until_stopped()
 
+    def restart_node(self, i, extra_args=None):
```

Comment:
@jnewbery added the option to override `node.extra_args`. I wonder if that this new `extra_arg` should be saved in `node`.

Reply:
Looks good. I don't think you need to update `node.extra_args`